library/common/api/external.cc

```diff
@@ -0,0 +1,25 @@
+#include "external.h"
+
+#include <unordered_map>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Api {
+namespace External {
+
+static std::unordered_map<std::string, void*> registry_{};
+
+void registerApi(std::string name, void* api) {
+  registry_[name] = api;
+}
+
+void* retrieveApi(std::string name) {
+  void* api = registry_[name];
+  ASSERT(api);
```

Comment:
Interesting, crash if not present?

Reply:
Yes, because I don't really see a readily-available better option, and at least for this case, this won't be exposed for direct user interaction (see the comment for more of my thinking on this).

If the API isn't located, a crash is probably about to happen anyways; this just does the operator of the library the courtesy of telling them why.